puppet/ha-puppet/screenshot.js

```diff
@@ -417,6 +417,7 @@ export class Browser {
 
       // If eink processing was requested, output PNG with specified colors
       if (einkColors) {
+	sharpInstance = sharpInstance.grayscale()
```

Comment:
It looks to me that it will now now do grayscale before going to the b-w color space. That doesn't seem good.

Reply:
It definitely does do greyscale before collapsing to the 1-bit `b-w` colorspace. 

In my testing that didn't seem to cause any issues or make the 1-bit/2-color images any worse (though they were already more or less unusable for my needs). I'm also not really sure what issues this would cause in the first place, though? I could _maybe_ see it interfering with the `threshold` call above, but that will have already happened before getting to this `grayscale` call (which should result in this being a NOOP for 1-bit outputs anyway).

I could instead move this to only apply to `eink` values other than `2` if you want?

Reply:
Ah, I dug deeper than the `sharp` docs into the underlying library and found this:

> `enum@Vips.Interpretation.B_W`: a monochrome image, roughly G from sRGB. The grey value is calculated in linear space with RGB ratios 0.2126, 0.7152, 0.0722 as defined by CIE 1931 linear luminance.

I think `grayscale` just sets all three channels to be the same value, but doesn't say how it chooses that. So maybe the right move is just to set the `b-w` colorspace on all `eink` levels instead of just `2`? I can try that when I get home tonight and follow up.